### PR TITLE
Add bean for CancerStudyPermissionEvaluator

### PR DESCRIPTION
**Pull request by Floris-Hyve**
Add a bean for `CancerStudyPermissionEvaluator` instead of instantiating directly within `createExpressionHandler`

### CODE REVIEW

File: src/main/java/org/cbioportal/security/CancerStudyPermissionEvaluator.java [hunk 1]
```diff
@@ -45,9 +45,11 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.*;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
```
Comment (haynescd) [inline src/main/java/org/cbioportal/security/CancerStudyPermissionEvaluator.java:48]: Can you remove unused imports

**Review by haynescd — CHANGES_REQUESTED** [review]

File: src/main/java/org/cbioportal/security/CancerStudyPermissionEvaluator.java (hunk 1 above)
Comment (dippindots) [reply src/main/java/org/cbioportal/security/CancerStudyPermissionEvaluator.java:48]: @Floris-Hyve I'm thinking the same thing, there are some unused imports, if we just need to change the way to create CancerStudyPermissionEvaluator, we don't need additional imports.

Comment (Floris-Hyve) [reply src/main/java/org/cbioportal/security/CancerStudyPermissionEvaluator.java]: Ah yes thanks for pointing out, removed them! This one seems unused also 
```import org.springframework.beans.factory.annotation.*;```

**Review by dippindots — APPROVED** [review]
LGTM

**Review by haynescd — APPROVED** [review]
LGTM

**Review by inodb — COMMENTED** [review]
@Floris-Hyve Thanks for trying to solve this! It's unclear why changing how the bean is instantiated would solve an authorization issue. I don't see any issue with merging this but would be good to first update title+description to explain the reasoning

**Review by inodb — CHANGES_REQUESTED** [review]
see comment re title/desciption

**Comment by Floris-Hyve** [comment]
I think there are issues with spring recognizing the `CancerStudyPermissionEvaluator` when it is not instantiated as a bean, which causes the `expressionHandler` to be set incorrectly. I think its related to how dependency injection works in spring. See below screenshots:
- Without instantiating as bean, study is accessible to a user without permission (can explore it too)
image

- As bean, study not accessible
image

**Pull request merged**